teamcity: failed tests on master: test/TestLogic, test/TestLogic/default, test/TestLogic/default/show_source

The following tests appear to have failed:

[#250465](https://teamcity.cockroachdb.com/viewLog.html?buildId=250465):

```
--- FAIL: test/TestLogic (29.360s)
test_log_scope.go:80: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic265796880
test_log_scope.go:63: use -show-logs to present logs inline
--- FAIL: test/TestLogic/default (0.110s)
null
--- FAIL: test/TestLogic/default/show_source (0.090s)
logic_test.go:1707: 
	testdata/logic_test/show_source:46: 
	expected:
	    name                                               current_value  type  description
	    diagnostics.reporting.interval                     1h0m0s         d     interval at which diagnostics data should be reported
	    diagnostics.reporting.report_metrics               true           b     enable collection and reporting diagnostic metrics to cockroach labs
	    diagnostics.reporting.send_crash_reports           true           b     send crash and panic reports
	    kv.allocator.lease_rebalancing_aggressiveness      1E+00          f     set greater than 1.0 to rebalance leases toward load more aggressively, or between 0 and 1.0 to be more conservative about rebalancing leases
	    kv.allocator.load_based_lease_rebalancing.enabled  true           b     set to enable rebalancing of range leases based on load and latency
	    kv.raft.command.max_size                           64 MiB         z     maximum size of a raft command
	    kv.raft_log.synchronize                            true           b     set to true to synchronize on Raft log writes to persistent storage
	    kv.snapshot_rebalance.max_rate                     2.0 MiB        z     the rate limit (bytes/sec) to use for rebalance snapshots
	    kv.snapshot_recovery.max_rate                      8.0 MiB        z     the rate limit (bytes/sec) to use for recovery snapshots
	    kv.transaction.max_intents                         100000         i     maximum number of write intents allowed for a KV transaction
	    server.declined_reservation_timeout                5s             d     the amount of time to consider the store throttled for up-replication after a reservation was declined
	    server.failed_reservation_timeout                  0s             d     the amount of time to consider the store throttled for up-replication after a failed reservation call
	    server.remote_debugging.mode                       local          s     set to enable remote debugging, localhost-only or disable (any, local, off)
	    server.time_until_store_dead                       5m0s           d     the time after which if there is no new gossiped information about a store, it is considered dead
	    sql.defaults.distsql                               1              e     Default distributed SQL execution mode [off = 0, auto = 1, on = 2]
	    sql.metrics.statement_details.dump_to_logs         false          b     dump collected statement statistics to node logs when periodically cleared
	    sql.metrics.statement_details.enabled              true           b     collect per-statement query statistics
	    sql.metrics.statement_details.threshold            0s             d     minmum execution time to cause statics to be collected
	    sql.trace.log_statement_execute                    false          b     set to true to enable logging of executed statements
	    sql.trace.session_eventlog.enabled                 false          b     set to true to enable session tracing
	    sql.trace.txn.enable_threshold                     0s             d     duration beyond which all transactions are traced (set to 0 to disable)
	    
	but found (query options: "colnames") :
	    name                                               current_value  type  description                                                                                                                                    
	    diagnostics.reporting.interval                     1h0m0s         d     interval at which diagnostics data should be reported                                                                                          
	    diagnostics.reporting.report_metrics               true           b     enable collection and reporting diagnostic metrics to cockroach labs                                                                           
	    diagnostics.reporting.send_crash_reports           true           b     send crash and panic reports                                                                                                                   
	    kv.allocator.lease_rebalancing_aggressiveness      1E+00          f     set greater than 1.0 to rebalance leases toward load more aggressively, or between 0 and 1.0 to be more conservative about rebalancing leases  
	    kv.allocator.load_based_lease_rebalancing.enabled  true           b     set to enable rebalancing of range leases based on load and latency                                                                            
	    kv.raft.command.max_size                           64 MiB         z     maximum size of a raft command                                                                                                                 
	    kv.raft_log.synchronize                            true           b     set to true to synchronize on Raft log writes to persistent storage                                                                            
	    kv.snapshot_rebalance.max_rate                     2.0 MiB        z     the rate limit (bytes/sec) to use for rebalance snapshots                                                                                      
	    kv.snapshot_recovery.max_rate                      8.0 MiB        z     the rate limit (bytes/sec) to use for recovery snapshots                                                                                       
	    kv.transaction.max_intents                         100000         i     maximum number of write intents allowed for a KV transaction                                                                                   
	    server.declined_reservation_timeout                5s             d     the amount of time to consider the store throttled for up-replication after a reservation was declined                                         
	    server.failed_reservation_timeout                  0s             d     the amount of time to consider the store throttled for up-replication after a failed reservation call                                          
	    server.remote_debugging.mode                       local          s     set to enable remote debugging, localhost-only or disable (any, local, off)                                                                    
	    server.time_until_store_dead                       5m0s           d     the time after which if there is no new gossiped information about a store, it is considered dead                                              
	    sql.defaults.distsql                               1              e     Default distributed SQL execution mode [auto = 1, on = 2, off = 0]                                                                             
	    sql.metrics.statement_details.dump_to_logs         false          b     dump collected statement statistics to node logs when periodically cleared                                                                     
	    sql.metrics.statement_details.enabled              true           b     collect per-statement query statistics                                                                                                         
	    sql.metrics.statement_details.threshold            0s             d     minmum execution time to cause statics to be collected                                                                                         
	    sql.trace.log_statement_execute                    false          b     set to true to enable logging of executed statements                                                                                           
	    sql.trace.session_eventlog.enabled                 false          b     set to true to enable session tracing                                                                                                          
	    sql.trace.txn.enable_threshold                     0s             d     duration beyond which all transactions are traced (set to 0 to disable)                                                                        
	
logic_test.go:1737: testdata/logic_test/show_source:72: too many errors encountered, skipping the rest of the input

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/settings/enum.go
+++ b/pkg/settings/enum.go
@@ -17,6 +17,7 @@ package settings
 import (
 	"bytes"
 	"fmt"
+	"sort"
 	"strconv"
 	"strings"
 
@@ -62,14 +63,18 @@ func (e *EnumSetting) set(k int64) error {
 
 func enumValuesToDesc(enumValues map[int64]string) string {
 	var buffer bytes.Buffer
+	values := make([]int64, 0, len(enumValues))
+	for k := range enumValues {
+		values = append(values, k)
+	}
+	sort.Slice(values, func(i, j int) bool { return values[i] < values[j] })
+
 	buffer.WriteString("[")
-	var notFirstElem bool
-	for k, v := range enumValues {
-		if notFirstElem {
+	for i, k := range values {
+		if i > 0 {
 			buffer.WriteString(", ")
 		}
-		fmt.Fprintf(&buffer, "%s = %d", strings.ToLower(v), k)
-		notFirstElem = true
+		fmt.Fprintf(&buffer, "%s = %d", strings.ToLower(enumValues[k]), k)
 	}
 	buffer.WriteString("]")
 	return buffer.String()

